wallet/src/lib.rs

```diff
@@ -57,14 +64,69 @@ impl WalletState {
         Self { utxos, pk_index }
     }
 
+    pub fn utxos_owned_by_pks(
+        &self,
+        pks: impl IntoIterator<Item = impl Borrow<PublicKey>>,
+    ) -> Vec<Utxo> {
+        pks.into_iter()
+            .filter_map(|pk| self.pk_index.get(pk.borrow()))
+            .flatten()
+            .map(|id| self.utxos[id])
+            .collect()
+    }
+
+    pub fn fund_tx<G: GasConstants>(
+        &self,
+        tx_builder: &MantleTxBuilder,
+        change_pk: PublicKey,
+        pks: impl IntoIterator<Item = impl Borrow<PublicKey>>,
```

Comment:
@danielSanchezQ the `Borrow` trait lets the caller decide if they want to consume or pass by reference. Best of both worlds!